tests/audiohelper.js

```diff
@@ -512,6 +520,31 @@ describe('AudioHelper', () => {
       });
 
       describe('setInputDevice', () => {
+        it('should wait for the beforeSetInputDevice to resolve', async () => {
+          const stub = sinon.stub();
+          audio = new AudioHelper(onActiveOutputsChanged, onActiveInputChanged, {
+            audioProcessorEventObserver: eventObserver,
+            beforeSetInputDevice: () => new Promise(res => res(stub())),
+            getUserMedia,
+            mediaDevices,
+            setSinkId: () => {}
+          });
+          await audio._setInputDevice('input');
```

Comment:
Let's use the public API
```suggestion
          await audio.setInputDevice('input');
```